### PR TITLE
Push files to container on install

### LINKED ISSUE #74: Operator may try to start the service before pushing the necessary files
### Bug Description

Currently we only push the necessary files to the container on pebble ready. This is not correct because the pebble_ready event may be fired after the relation_joined events.

This is easy to reproduce when scaling up

### To Reproduce

1. juju deploy kratos
2. juju integrate kratos postgres
3. juju integrate kratos:admin_ingress traefik-admin
4. juju integrate kratos:public_ingress traefik-public
5. juju add-unit kratos -n 3

### Environment

-

### Relevant log output

```shell
-
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nsklikas**
Push files to container on install, this makes sure that they will be pushed before any hook that tries to run the service.

Closes #74

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -396,6 +399,16 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
...
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Kratos container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Kratos container")
+            return
+
+        self._push_default_files()
+
+
     def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
```
Comment (nsklikas) [inline src/charm.py:412]: Seeing how the charm code starts to turn out, I am not sure if we need this function to be different from `_handle_status_update_config`

File: src/charm.py [hunk 2]
```diff
@@ -375,58 +376,42 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
...
+            return
+
+        self._container.push(self._config_file_path, self._render_conf_file(), make_dirs=True)
+        # We need to push the layer because this may run before _on_pebble_ready
+        try:
             self._container.restart(self._container_name)
-            self.unit.status = ActiveStatus()
+        except ChangeError as err:
+            logger.error(str(err))
+            self.unit.status = BlockedStatus("Failed to replan, please consult the logs")
```
Comment (natalian98) [inline src/charm.py:398]:
```suggestion
            self.unit.status = BlockedStatus("Failed to restart, please consult the logs")
```

File: src/charm.py [hunk 3]
```diff
@@ -375,58 +376,42 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
...
 
-        if self.model.relations[self._db_relation_name]:
-            self.unit.status = WaitingStatus("Waiting for database creation")
-        else:
-            self.unit.status = BlockedStatus("Missing postgres database relation")
+        self.unit.status = ActiveStatus()
 
-    def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
-        """Event Handler for pebble ready event."""
+    def _on_install(self, event):
```
Comment (natalian98) [inline src/charm.py:403]:
```suggestion
    def _on_install(self, event: InstallEvent) -> None:
```

File: tests/integration/test_charm.py [hunk 4]
```diff
@@ -253,3 +259,25 @@ async def test_identity_schemas_config(ops_test: OpsTest) -> None:
     resp = requests.get(f"http://{public_address}/{ops_test.model.name}-{APP_NAME}/schemas")
 
     assert original_resp == resp.json()
+
+
+async def test_kratos_scale_up(ops_test: OpsTest):
```
Comment (natalian98) [inline tests/integration/test_charm.py:264]:
```suggestion
async def test_kratos_scale_up(ops_test: OpsTest) -> None:
```

**Review by natalian98 — COMMENTED** [review]
Looks good, I have some minor comments.

**Review by natalian98 — APPROVED** [review]

**Pull request merged**